packages/core/content-manager/admin/src/pages/ListView/components/BulkActions/Actions.tsx

```diff
@@ -292,195 +301,193 @@ const BulkActionModal = ({
+                )}
+              </Radio>
+              <Radio
+                checked={isDeleteAllLocale}
+                value={BULK_DELETE_OPTIONS.ALL_LOCALES}
+                name="bulk-delete-options"
+              >
+                {formatMessage({
+                  id: 'popUpWarning.bodyMessage.contentType.delete.allLocales',
+                  defaultMessage: 'Delete entries (all locales)',
+                })}
+              </Radio>
+            </Flex>
+          </Box>
+        </Flex>
+      ),
+      onConfirm: handleConfirmBulkDelete,
+    },
+  };
+};
+
+const UnpublishAction: BulkActionComponent = ({ documentIds, model }) => {
+  const { formatMessage } = useIntl();
+  const { schema } = useDoc();
+  const { selectRow, selectedRows } = useTable('unpublishAction', (state) => state);
+  const hasPublishPermission = useDocumentRBAC('unpublishAction', (state) => state.canPublish);
+  const hasI18nEnabled = Boolean(schema?.pluginOptions?.i18n);
+  const { unpublishMany: bulkUnpublishAction } = useDocumentActions();
+
+  const handleConfirmBulkUnpublish = async () => {
+    const data = await bulkUnpublishAction({ documentIds, model });
+    if (!('error' in data)) {
+      selectRow([]);
+    }
+  };
+
+  const showUnpublishButton =
+    hasPublishPermission && selectedRows.some((entry) => entry.status === 'published');
+
+  if (!showUnpublishButton) return null;
+
+  return {
+    actionType: 'unpublish',
+    variant: 'tertiary',
+    label: formatMessage({ id: 'app.utils.unpublish', defaultMessage: 'Unpublish' }),
+    dialog: {
+      type: 'dialog',
+      title: formatMessage({
+        id: 'app.components.ConfirmDialog.title',
+        defaultMessage: 'Confirmation',
+      }),
+      content: (
+        <Flex direction="column" alignItems="stretch" gap={2}>
+          <Typography id="confirm-description" textAlign="center">
+            {formatMessage({
+              id: 'popUpWarning.bodyMessage.contentType.unpublish.all',
+              defaultMessage: 'Are you sure you want to unpublish these entries?',
+            })}
+          </Typography>
+          {hasI18nEnabled && (
+            <Box textAlign="center" padding={3}>
+              <Typography textColor="danger500">
+                {formatMessage(
+                  {
+                    id: getTranslation('Settings.list.actions.unpublishAdditionalInfos'),
+                    defaultMessage:
+                      'This will unpublish the active locale versions <em>(from Internationalization)</em>',
+                  },
+                  {
+                    em: Emphasis,
+                  }
+                )}
+              </Typography>
+            </Box>
+          )}
+        </Flex>
+      ),
+      confirmButton: formatMessage({
+        id: 'app.utils.unpublish',
+        defaultMessage: 'Unpublish',
+      }),
+      onConfirm: handleConfirmBulkUnpublish,
+    },
+  };
+};
+
+export const getContentPermissions = (subject: string) => {
```

Comment:
Why is this exported? It's not used in this file, i'm wondering if we can get rid of it, it's a bit weird.

Reply:
absolutely..